### DISCUSSION TITLE
如何让某些局域网地址通过内核？

### DISCUSSION THREAD

**Discussion by imbytecat** [kkkgo/PaoPaoGateWay#120] [问和答]
我的使用场景是，除了日常的科学上网分流，我另外还加了家里开出来的节点，方便我直接访问家里的设备，PaoPaoGateway 是不是默认不会让局域网地址通过？

使用 OpenClash 的话。会有这个绕过内核的 iptables 配置项，不知道 PPGW 有没有提供呢？

**Comment by kkkgo** [comment]
如果是配合PaoPaoDNS使用，使用域名进行连接并加入force_forward_list.txt，就会解析为FAKEIP，自行配置规则走你的对应节点即可，理论上PaoPaoGateWay并不关心你实际访问的是不是局域网地址或者是什么地址甚至是IPv6内容，只要你是域名，送往节点的就是域名，送回来也是域名。如果没有域名，可以直接用nip.io，比如192.168.1.1.nip.io。
如果是想以裸IP形式访问，建议用zerotier等三层组网VPN加静态路由。

**Comment by mustang5910** [comment]
恰好遇到了类似的需求, 我这边的情况是有一个socks代理能访问远端的 10.0.0.0/8 网段, 希望通过静态路由 10.0.0.0/8 到 ppgw，来实现无感的组网ip访问。 默认的 ppgw 不会路由局域网网段， 需要一定程度的修改:

1. 克隆项目到本地
2. 修改 FILES/usr/bin/nft.sh 和 FILES/usr/bin/nft_tcp.sh， 修改 localnetwork 中对应的网段， 对我来说是移除 10.0.0.0/8 这一行
3. 运行 makeiso.sh 脚本， 即可生成新的镜像文件
4. 按照正常流程配置socks代理和主路由的静态路由
5. happy working

**Comment by kkkgo** [comment]
本地编译会开启ssh调试服务。
已经新增定制功能：Docker定制network.ini添加自定义局域网属性地址功能，详情参考文档自定义局域网属性地址一节。
